### PR TITLE
Feature/appcli 91 add pre commit hook for checking hadolint rules

### PR DESCRIPTION
**Pull request by benjamin-wiffen**
This is ready to be merged.

To install the new pre-commit hooks, clone / pull to a local repo and run either make pre-commit or make all.
This affects the .git folder of the LOCAL INSTANCE ONLY,  so the make pre-commit needs to be re-run every time the repo is cloned.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -67,6 +67,9 @@ format: venv
 format-check: venv
 	${PYTHON} -m black . --diff --check
 
+pre-commit: venv
+	pre-commit install --config .github/.pre-commit-config.yaml
```
Comment (thomas-anderson-bsl) [inline Makefile:71]: The goal when installing pre-commit hooks is to do it as early as possible (i.e. as soon as we can after installing the `pre-commit` with `pip install`), and to run the task whenever the `.github/.pre-commit-config.yaml` file is changed.

If you look at line 43, we do a check (when running `venv` target, which depends on `$(VENV_NAME)/bin/activate`) that `setup.py` has been changed. 

I believe the best option would be to add a dependency to `venv` on `.github/.pre-commit-config.yaml` which will re-run the full installation, as well as run `pre-commit install`

File: Makefile [hunk 2]
```diff
@@ -89,6 +92,6 @@ docker-publish: docker
 	docker push brightsparklabs/appcli:${APP_VERSION}
 	docker push brightsparklabs/appcli:latest
 
-all: format isort lint test
+all: format isort lint test pre-commit
```
Comment (thomas-anderson-bsl) [inline Makefile:95]: This won't be necessary if we run `pre-commit install` within the `venv` block, as described above

File: Makefile (hunk 1 above)
Comment (thomas-anderson-bsl) [reply Makefile:71]: This will also mean this make target can be deleted

**Review by thomas-anderson-bsl — CHANGES_REQUESTED** [review]
Requesting slight changes to how we run the install, so it's done earlier in the make run cycle and more reliably

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,11 +12,12 @@ The changelog is applicable from version `1.0.0` onwards.
 
 ### Breaking Changes from 1.3.6
 
-As a result of supporting application-level settings files, all references to
-settings in template files have moved. See the README for details on migration.
+- Either `make all` or `make pre-commit` needs to be run when this repo is cloned to setup the pre-commit hooks.
```
[the remark this answers is not in the document]
Comment (benjamin-wiffen) [reply CHANGELOG.md]: Removed

File: Makefile (hunk 1 above)
Comment (benjamin-wiffen) [reply Makefile]: This has been moved up in to `venv`.

File: Makefile (hunk 2 above)
Comment (benjamin-wiffen) [reply Makefile]: Removed

**Review by thomas-anderson-bsl — APPROVED** [review]

**Pull request merged**